src/test/java/uk/gov/pay/products/resources/PaymentResourceIT.java

```diff
@@ -36,6 +37,45 @@ public class PaymentResourceIT extends IntegrationTest {
     private final String nextUrl = "www.gov.uk/pay";
     private final int gatewayAccountId = randomInt();
 
+    @Test
+    public void redactPaymentReference() {
+        ProductEntity productEntity = ProductEntityFixture.aProductEntity()
+                .withGatewayAccountId(gatewayAccountId)
+                .withReferenceEnabled(true)
+                .withReferenceLabel("Reference label")
+                .withLanguage(SupportedLanguage.WELSH)
+                .build();
+
+        Product product = productEntity.toProduct();
+
+        databaseHelper.addProduct(product);
+
+        Integer productId = databaseHelper.findProductId(productEntity.getExternalId());
+        productEntity.setId(productId);
+        
+        PaymentEntity payment = uk.gov.pay.products.fixtures.PaymentEntityFixture.aPaymentEntity()
+                .withExternalId(randomUuid())
+                .withStatus(uk.gov.pay.products.util.PaymentStatus.CREATED)
+                .withGovukPaymentId("kts8ici6rm")
+                .withProduct(productEntity)
+                .withReferenceNumber("4242424242424242")
+                .build();
+        databaseHelper.addPayment(payment.toPayment(), 1);
```

Comment:
Maybe add another payment with a different ID so we can check the redaction process does not go rogue and redact everything!

Reply:
[Done](https://github.com/alphagov/pay-products/pull/1027/commits/a7e5cbf0cc85d728caa4fdee3aea86db232d47c0#diff-e67433f0ac4091fb920867ef85e9b88d5bfff5e0b465462523b90c7d4f77cd50R86-R89).